vault/resource_ad_secret_library.go

```diff
@@ -0,0 +1,232 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/vault/api"
+	"github.com/terraform-providers/terraform-provider-vault/util"
+)
+
+var (
+	adSecretBackendFromLibraryPathRegex = regexp.MustCompile("^(.+)/library/.+$")
+	adSecretBackendSetNameFromPathRegex = regexp.MustCompile("^.+/library/(.+$)")
+)
+
+func adSecretBackendLibraryResource() *schema.Resource {
+	fields := map[string]*schema.Schema{
+		"backend": {
+			Type:        schema.TypeString,
+			Required:    true,
+			ForceNew:    true,
+			Description: "The mount path for the AD backend.",
+			StateFunc: func(v interface{}) string {
+				return strings.Trim(v.(string), "/")
+			},
+		},
+		"name": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: `The name of the set of service accounts.`,
+			ForceNew:    true,
+		},
+		"service_account_names": {
+			Type:        schema.TypeList,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			Required:    true,
+			Description: "The names of all the service accounts that can be checked out from this set. These service accounts must already exist in Active Directory.",
+			ForceNew:    true,
+		},
+		"max_ttl": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Computed:    true,
+			Description: `The maximum amount of time a check-out last with renewal before Vault automatically checks it back in. Defaults to 24 hours. Setting it to zero reflects an unlimited lending period.`,
+		},
+		"ttl": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Computed:    true,
+			Description: `The maximum amount of time a single check-out lasts before Vault automatically checks it back in. Defaults to 24 hours. Setting it to zero reflects an unlimited lending period.`,
+		},
```

Comment:
Are these the same parameter? They have almost identical descriptions.

And since these are TypeInt, might want to mention in the description that input should be in seconds (since the [API apparently accepts a duration string](https://www.vaultproject.io/api/secret/ad#library-management)).

Reply:
Good catch, done!